src/v/cluster/members_manager.cc

```diff
@@ -327,6 +326,21 @@ members_manager::apply_raft_configuration_batch(model::record_batch b) {
     co_return make_error_code(errc::success);
 }
 
+ss::future<> members_manager::fill_snapshot(controller_snapshot& snap) const {
+    snap.members.node_ids_by_uuid = _id_by_uuid;
+    snap.members.next_assigned_id = _next_assigned_id;
+    co_return;
+}
+
+ss::future<> members_manager::apply_snapshot(
+  const raft::snapshot_metadata& meta, const controller_snapshot& snap) {
+    _id_by_uuid = snap.members.node_ids_by_uuid;
+    _next_assigned_id = snap.members.next_assigned_id;
+    co_await handle_raft0_cfg_update(
+      meta.latest_configuration, meta.last_included_index);
```

Comment:
this part is missing nodes that were removed during the cluster lifecycle

Reply:
i think that we should store all necessary information in the snapshot itself and not relaying on configuration from `snapshot_metadata`